wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java

```diff
@@ -164,6 +202,89 @@ private String getTargetHPCCBuildVersionString() throws Exception
 
     }
 
+    public SpanBuilder getSpanBuilder(String spanName)
+    {
+        Tracer tracer = getTracer(INSTRUMENTED_LIBRARY_NAME);
+        SpanBuilder spanBuilder = tracer.spanBuilder(spanName)
+          .setAttribute(ServerAttributes.SERVER_ADDRESS, wsconn.getHost())
+          .setAttribute(ServerAttributes.SERVER_PORT, Long.getLong(wsconn.getPort()))
+          .setAttribute(HttpAttributes.HTTP_REQUEST_METHOD, HttpAttributes.HttpRequestMethodValues.GET)
+          .setSpanKind(SpanKind.CLIENT);
+        //.startSpan();
+
+        return spanBuilder;
+    }
+
+    public void injectTraceParentHeader(Span currentSpan)
+    {
+        injectTraceParentHeader(stub._getServiceClient().getOptions(), currentSpan);
+    }
+
+    static public void injectTraceParentHeader(Options options, Span currentSpan)
+    {
+        if (options != null)
+        {
+            if (currentSpan != null && currentSpan.getSpanContext().isValid())
+                W3CTraceContextPropagator.getInstance().inject(Context.current(), options, Options::setProperty);
+        }
+    }
+
+    static public void injectCurrentSpanTraceParentHeader(Stub clientStub)
+    {
+        if (clientStub != null)
+        {
+            injectCurrentSpanTraceParentHeader(clientStub._getServiceClient().getOptions());
+        }
+    }
+
+    static public void injectCurrentSpanTraceParentHeader(Options options)
+    {
+        if (options != null)
+        {
+            injectTraceParentHeader(options, Span.current());
+        }
+    }
+
+    static public String getTraceParentHeader()
+    {
+        String traceparent = null;
+        Span currentSpan = Span.current();
+        if (currentSpan != null && currentSpan.getSpanContext().isValid())
+        {
+            Map<String, String> carrier = new HashMap<>();
+            TextMapSetter<Map<String, String>> setter = Map::put;
+            W3CTraceContextPropagator.getInstance().inject(Context.current(), carrier, setter);
```

Comment:
Should we be using currentSpawn.getContext() here as well?

Reply:
inject actually takes a Context rather than SpanContext